metashare/api/serializers.py

```diff
@@ -63,3 +78,33 @@ class Meta:
 
     def get_branch_url(self, obj):
         return f"{obj.product.repo_url}/tree/{obj.branch_name}"
+
+
+class TaskSerializer(serializers.ModelSerializer):
+    id = serializers.CharField(read_only=True)
+    description = MarkdownField(allow_blank=True)
+    project = serializers.PrimaryKeyRelatedField(
+        queryset=Project.objects.all(), pk_field=serializers.CharField()
+    )
+    assignee = HashidPrimaryKeyRelatedField(
+        queryset=User.objects.all(), allow_null=True
+    )
+
+    class Meta:
+        model = Task
+        fields = (
+            "id",
+            "name",
+            "description",
+            "project",
+            "assignee",
+            "slug",
+            "old_slugs",
+        )
+        validators = (
+            UniqueTogetherValidator(
+                queryset=Task.objects.all(),
+                fields=("name", "project"),
+                message=_("A task with this name already exists."),
```

Comment:
@wlonk Similar to projects, I think we want to disallow names that are the same, case-insensitive. Right now I can add `task` and `Task` as separate items.

Reply:
Yeah. That ol' case.